test/gateway_composer/request_stake.js

```diff
@@ -12,38 +12,332 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const BN = require('bn.js');
 const { AccountProvider } = require('../test_lib/utils.js');
 
 const utils = require('../test_lib/utils');
 const gatewayComposerUtils = require('./utils');
 
-contract('GatewayComposer::requestStake', async () => {
-    contract('Positive Tests', async (accounts) => {
+
+contract('GatewayComposer::requestStake', async (accounts) => {
+    describe('Negative Tests', async () => {
         const accountProvider = new AccountProvider(accounts);
 
-        it('Returns stake request hash.', async () => {
+        it('Fails when msg.sender is not owner.', async () => {
+            const {
+                valueToken,
+                gatewayComposer,
+                owner,
+                brandedToken,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const {
+                stakeAmount,
+            } = await gatewayComposerUtils.setupGatewayComposerRequestStake(
+                valueToken,
+                gatewayComposer,
+                owner,
+            );
+
+            const mintAmount = await brandedToken.convertToBrandedTokens(stakeAmount);
+            const gateway = accountProvider.get();
+            const beneficiary = accountProvider.get();
+            const gasPrice = 1;
```

Comment:
Refactoring comment: suggest declaring `gasprice`, `gaslimit`, and `nonce` above the `it`s, as they are invariant and do not impact the tests.

Reply:
Same can be done for gateway and beneficiary also